operator/v1/types_ingress.go

```diff
@@ -672,25 +672,27 @@ type IngressControllerCaptureHTTPCookie struct {
 	// name specifies a cookie name.  Its value must be a valid HTTP cookie
 	// name as defined in RFC 6265 section 4.1.
 	//
-	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
-	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]*$"
+	// +kubebuilder:validation:MinLength=0
 	// +kubebuilder:validation:MaxLength=1024
 	// +optional
 	Name string `json:"name"`
 
 	// namePrefix specifies a cookie name prefix.  Its value must be a valid
 	// HTTP cookie name as defined in RFC 6265 section 4.1.
 	//
-	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
-	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]*$"
+	// +kubebuilder:validation:MinLength=0
 	// +kubebuilder:validation:MaxLength=1024
 	// +optional
 	NamePrefix string `json:"namePrefix"`
 
-	// maxLength specifies a maximum length for the cookie value.  If a
-	// cookie value exceeds this length, the value will be truncated in the
-	// log message.  Note that the ingress controller may impose a separate
-	// bound on the total length of HTTP headers in a request.
+	// maxLength specifies a maximum length of the string that will be
```

Comment:
actually, why is this required.  Seems like we could choose a reasonable default max length

Reply:
I guess we could, but it would be pretty arbitrary because we do not know the cookie's purpose.  Would 256 be a reasonable default?

Reply:
Does the minimum need to be set to 0 in order for validation to admit an object that does not specify the value?